wandb/sdk/wandb_run.py

```diff
@@ -1133,9 +1146,14 @@ def _config_callback(
             self._backend.interface.publish_config(key=key, val=val, data=data)
 
     def _config_artifact_callback(
-        self, key: str, val: Union[str, Artifact]
+        self, key: str, val: Union[str, Artifact, dict]
     ) -> Union[Artifact, public.Artifact]:
-        if _is_artifact_string(val):
+        if _is_artifact_dict(val):
+            assert isinstance(val, dict)
+            public_api = self._public_api()
```

Comment:
could leave some documentation about how the artifact_dict would look like?
(i see that in the string case we also have some address case)